projectq/ops/_gates.py

```diff
@@ -338,3 +338,59 @@ def get_inverse(self):
 
 #: Shortcut (instance of) :class:`projectq.ops.BarrierGate`
 Barrier = BarrierGate()
+
+
+class FlipBits(SelfInverseGate):
+    """ Gate for transforming qubits in state |0> to any computational basis state """
+    def __init__(self, bits_to_flip):
+        """
+        Initialize FlipBits gate.
+
+        Example:
+            .. code-block:: python
+
+                qureg = eng.allocate_qureg(2)
+                FlipBits([0, 1]) | qureg
+
+        Note:
+            The amplitude of state k is final_state[k]. When the state k is
+            written in binary notation, then qureg[0] denotes the qubit
+            whose state corresponds to the least significant bit of k.
+
+        Args:
+            bits_to_flip(list[int]|list[bool]|str|int): int or array of 0/1, True/False, or
+                                    string of 0/1 of length up to len(qureg) identifying
+                                    the qubits to flip. In case of int, the bits to flip are
+                                    determined from the binary digits, with the least significant
+                                    bit corresponding to qureg[0]. If bits_to_flip is negative,
+                                    exactly all bits which would not be flipped for the input
+                                    -bits_to_flip are flipped.
+        """
+        if isinstance(bits_to_flip, int):
+            self.bits_to_flip = bits_to_flip
+        elif isinstance(bits_to_flip, str):
+            self.bits_to_flip = list([ c != "0" for c in bits_to_flip])
+        else:
+            self.bits_to_flip = list(bits_to_flip)
+
+    def __str__(self):
+        return "FlipBits("+str(self.bits_to_flip)+")"
+
+    def __or__(self, qubits):
+        for qureg in self.make_tuple_of_qureg(qubits):
+            if isinstance(self.bits_to_flip, int):
+                bits_to_flip = [((self.bits_to_flip if self.bits_to_flip >= 0 else self.bits_to_flip-1) >> i) & 1 for i in range(len(qureg))]
+            else:
+                bits_to_flip = self.bits_to_flip
+
+            for i, flip in enumerate(bits_to_flip):
+                if flip:
+                    XGate() | qureg[i]
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return self.bits_to_flip == other.bits_to_flip
```

Comment:
With fixed storage type, we also don't have strange behavior of `__eq__` just because we initialized it with an int as opposed to a list of bits.